examples/seeder_example/vt_application.cpp

```diff
@@ -0,0 +1,1149 @@
+	std::string objectPoolHash = isobus::IOPFileInterface::hash_object_pool_to_version(objectPool);
+
+	VTClientInterface.set_object_pool(0, isobus::VirtualTerminalClient::VTVersion::Version4, objectPool.data(), objectPool.size(), objectPoolHash);
+	softkeyEventListener = VTClientInterface.add_vt_soft_key_event_listener([this](const isobus::VirtualTerminalClient::VTKeyEvent &event) { this->handle_vt_key_events(event); });
+	buttonEventListener = VTClientInterface.add_vt_button_event_listener([this](const isobus::VirtualTerminalClient::VTKeyEvent &event) { this->handle_vt_key_events(event); });
+	numericValueEventListener = VTClientInterface.add_vt_change_numeric_value_event_listener([this](const isobus::VirtualTerminalClient::VTChangeNumericValueEvent &event) { this->handle_numeric_value_events(event); });
+	VTClientInterface.initialize(true);
+
+	for (std::uint32_t i = 0; i < static_cast<std::uint32_t>(UpdateVTStateFlags::NumberOfFlags); i++)
+	{
+		txFlags.set_flag(i); // Set all flags to bring the pool up with a known state
+	}
+
+	speedMessages.initialize();
+	machineSelectedSpeedEventHandle = speedMessages.get_machine_selected_speed_data_event_publisher().add_listener([this](const std::shared_ptr<isobus::SpeedMessagesInterface::MachineSelectedSpeedData> mssData, bool changed) { this->handle_machine_selected_speed(mssData, changed); });
+	groundBasedSpeedEventHandle = speedMessages.get_ground_based_machine_speed_data_event_publisher().add_listener([this](const std::shared_ptr<isobus::SpeedMessagesInterface::GroundBasedSpeedData> gbsData, bool changed) { this->handle_ground_based_speed(gbsData, changed); });
+	wheelBasedSpeedEventHandle = speedMessages.get_wheel_based_machine_speed_data_event_publisher().add_listener([this](const std::shared_ptr<isobus::SpeedMessagesInterface::WheelBasedMachineSpeedData> wbsData, bool changed) { this->handle_wheel_based_speed(wbsData, changed); });
+
+	ddop = std::make_shared<isobus::DeviceDescriptorObjectPool>(3);
+	if (sectionControl.create_ddop(ddop, TCClientInterface.get_internal_control_function()->get_NAME()))
+	{
+		TCClientInterface.configure(ddop, 1, 255, 255, true, true, true, false, true);
+		TCClientInterface.add_request_value_callback(SectionControlImplementSimulator::request_value_command_callback, &sectionControl);
+		TCClientInterface.add_value_command_callback(SectionControlImplementSimulator::value_command_callback, &sectionControl);
+		TCClientInterface.initialize(true);
+	}
+	else
+	{
+		std::cout << "Failed generating DDOP. TC functionality will not work until the DDOP structure is fixed." << std::endl;
+		retVal = false;
+	}
+	return retVal;
+}
+
+void SeederVtApplication::handle_vt_key_events(const isobus::VirtualTerminalClient::VTKeyEvent &event)
+{
+	switch (event.keyEvent)
+	{
+		case isobus::VirtualTerminalClient::KeyActivationCode::ButtonUnlatchedOrReleased:
+		{
+			switch (event.objectID)
+			{
+				case home_Key:
+				{
+					set_currently_active_screen(ActiveScreen::Main);
```

Comment:
I feel like we should prefer the `get_visible_data_mask()` getter for checking if a screen is active instead, there could perhaps be some proprietary method of switching data masks, though unlikely, you never know haha

Reply:
I could try doing that. I guess the idea was that if we got disconnected from the VT that there'd be a way to go back to the last active screen by setting the flag for updating the data mask instead of always ending up at the default data mask in the pool. Doesn't really matter much for this simple example but in a larger pool it might be nice, IDK

Reply:
Hmm, irl I probably would prefer to being reset to the main page instead. I mean what would be the odds of getting disconnected without it being on purpose? Maybe the use cases on my machines are not that diverse, but still idk if it's worth the hassle of keeping our own `active_screen` tracking system